core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesShard.scala

```diff
@@ -292,19 +293,21 @@ class DownsampledTimeSeriesShard(rawDatasetRef: DatasetRef,
         // since it is the highest resolution/ttl
         downsampleTtls.last.toMillis -> downsampledDatasetRefs.last
       case TimeRangeChunkScan(startTime, _) =>
-        val ttlIndex = downsampleTtls.indexWhere(t => startTime > System.currentTimeMillis() - t.toMillis)
+        var ttlIndex = downsampleTtls.indexWhere(t => startTime > System.currentTimeMillis() - t.toMillis)
```

Comment:
I don't understand this change.

Reply:
This picks the downsample resolution based on the earliest time of the query. If earliest time of query is older than any retention then -1 is returned. Then we have to resort to the highest resolution with highest TTL

Reply:
Sounds good. Please add this comment to the code.